pkg/brokerpak/cmd.go

```diff
@@ -148,10 +151,30 @@ func Validate(pack string) error {
 // RegisterAll fetches all brokerpaks from the settings file and registers them
 // with the given registry.
 func RegisterAll(registry broker.BrokerRegistry) error {
-	localPacks := viper.GetStringSlice("brokerpak.local_packs")
+	registerLogger := utils.NewLogger("brokerpak-registration")
+	// create a temp directory to hold all the paks
+	pakDir, err := ioutil.TempDir("", "brokerpaks")
+	if err != nil {
+		return fmt.Errorf("couldn't create brokerpak staging area: %v", err)
+	}
+	defer os.RemoveAll(pakDir)
+
+	for i, packSource := range viper.GetStringSlice("brokerpak.packs") {
+		destFile := filepath.Join(pakDir, fmt.Sprintf("pack-%d.brokerpak", i))
+		registerLogger.Debug("importing brokerpak", lager.Data{
+			"source":      packSource,
+			"destination": destFile,
+		})
+
+		if err := fetchBrokerpak(packSource, destFile); err != nil {
```

Comment:
One potential optimization would be to implement a fan-out/fan-in parallel fetches here, but I wouldn't bother unless this actually becomes a bottleneck.

Reply:
I added a note inline to keep track of this potential improvement.